lib/doctor/DiagnosticInfoBuilder.ts

```diff
@@ -0,0 +1,214 @@
+import { getProjectConfig } from '../projects';
+import { fetchProject } from '@hubspot/local-dev-lib/api/projects';
+import path from 'path';
+import pkg from '../../package.json';
+import { logger } from '@hubspot/local-dev-lib/logger';
+import { Environment } from '@hubspot/local-dev-lib/types/Config';
+import {
+  AccessToken,
+  AccountType,
+  AuthType,
+} from '@hubspot/local-dev-lib/types/Accounts';
+import { Project } from '@hubspot/local-dev-lib/types/Project';
+import { getAccountId } from '@hubspot/local-dev-lib/config';
+import { getAccountConfig, getConfigPath } from '@hubspot/local-dev-lib/config';
+import { getAccessToken } from '@hubspot/local-dev-lib/personalAccessKey';
+import { walk } from '@hubspot/local-dev-lib/fs';
+import util from 'util';
+import { exec as execAsync } from 'child_process';
+import process from 'process';
+
+export type ProjectConfig = Awaited<ReturnType<typeof getProjectConfig>>;
+
+const hubspotCli = '@hubspot/cli';
+interface FilesInfo {
+  files: string[];
+  configFiles: string[];
+  packageFiles: string[];
+  packageLockFiles: string[];
+  envFiles: string[];
+  jsonFiles: string[];
+}
+
+export interface DiagnosticInfo extends FilesInfo {
+  path?: string;
+  versions: { [hubspotCli]: string; node: string; npm: string | null };
+  config: string | null;
+  project: {
+    details?: Project;
+    config?: ProjectConfig;
+  };
+  arch: typeof process.arch;
+  platform: typeof process.platform;
+  account: {
+    name?: string;
+    accountId?: number | null;
+    scopeGroups?: AccessToken['scopeGroups'];
+    enabledFeatures?: AccessToken['enabledFeatures'];
+    accountType?: AccessToken['accountType'];
+    authType?: AuthType;
+  };
+  diagnosis?: string;
+  errorCount?: number;
+  warningCount?: number;
+}
+
+const configFiles = [
+  'serverless.json',
+  'hsproject.json',
+  'app.json',
+  'public-app.json',
+  'theme.json',
+  'cms-assets.json',
+];
+
+export class DiagnosticInfoBuilder {
+  accountId: number | null;
+  readonly env?: Environment;
+  readonly authType?: AuthType;
+  readonly accountType?: AccountType;
+  readonly personalAccessKey?: string;
+  private _projectConfig?: ProjectConfig;
+  private accessToken?: AccessToken;
+  private projectDetails?: Project;
+  private files?: string[];
+  readonly processInfo: NodeJS.Process;
+
+  constructor(processInfo: NodeJS.Process) {
+    this.accountId = getAccountId();
+    const accountConfig = getAccountConfig(this.accountId!);
+    this.env = accountConfig?.env;
+    this.authType = accountConfig?.authType;
+    this.accountType = accountConfig?.accountType;
+    this.personalAccessKey = accountConfig?.personalAccessKey;
+    this.processInfo = processInfo;
+  }
+
+  async generateDiagnosticInfo(): Promise<DiagnosticInfo> {
+    this._projectConfig = await getProjectConfig(null);
```

Comment:
nit: i think it would be a little clearer to just do `getProjectConfig()`. this is done a few other times in the repo

Reply:
I'm getting a TS error if I omit the arg since it's not optional.  I can just add a `// @ts-expect-error` so when this method is type that can be deleted

Reply:
Ohh weird, I didn't realized TS had any opinions about stuff defined in `// @ts-nocheck` files. Fine by me either way then